src/dropdown/Dropdown.d.ts

```diff
@@ -39,19 +35,19 @@ export type DropdownProps = {
   autoBlur?: boolean,
   simpleValue?: boolean,
   menuPosition?: 'top' | 'bottom',
-  menuRenderer?: (menuItem: object) => any, // TODO: t.ReactChildren
+  menuRenderer?: (menuItem: DropdownMenuRendererProps) => React.ReactElement<any>, // TODO: t.ReactChildren
   groupByKey?: string,
-  optionGroupRenderer?: (title: string) => any, // TODO: t.ReactChildren
-  placeholder?: JSX.Element | string, // TODO: t.maybe(t.union([t.String, t.ReactElement]))
-  noResultsText?: JSX.Element | string,
+  optionGroupRenderer?: (title: string) => JSX.Element, // TODO: t.ReactChildren
+  placeholder?: string | React.ReactElement<any>, // TODO: t.maybe(t.union([t.String, t.ReactElement]))
+  noResultsText?: string,
   allowCreate?: boolean,
-  addLabelText?: JSX.Element | string,
-  valueRenderer?: (option: object) => JSX.Element, // TODO: t.ReactChildren
-  optionRenderer?: (option: object) => JSX.Element, // TODO: t.ReactChildren
+  addLabelText?: string,
+  valueRenderer?: (option: Option) => JSX.Element, // TODO: t.ReactChildren
+  optionRenderer?: (option: Option) => JSX.Element, // TODO: t.ReactChildren
   delimiter?: string,
-  onInputChange?: () => void,
-  onFocus?: () => void,
-  onBlur?: () => void,
+  onInputChange?: (inputValue: string) => void,
+  onFocus?: FocusEventHandler<{}>,
```

Comment:
I know you copy-pasted this from `@types/react-select`, but I think we should pass an HTML element to `FocusEventHandler` as we do for every other `SyntheticEvent` like: `FocusEventHandler<HTMLDivElement>` (if it's a div or simply `HTMLElement` in the other case)